### PR TITLE
Allow to read the model_type value from a checkpoint file

### LINKED ISSUE #335: Allow to read the model_type value from a checkpoint file
Allow to read the model_type value from a checkpoint (dnn) file, as opposed to only from a .json file in a model directory.

### PR DESCRIPTION
**Pull request by franzmpub**
# Pull Request      

## What does this PR do?

```Closes #335```       
 
## Description

```Allow to read the "model_type" value from a checkpoint file, as opposed to only from a .json file in a model directory. Also backwards compatible with reading the model type from  "model" in "arguments" in a  checkpoint file```

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [ ] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [x] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [x] Added [description](#description) of changes proposed.
- [x] Review requested as [appropriate](#request-review).
- [x] Version bumped as [appropriate](#Versioning).
- [x] New classes, methods, and functions documented.
- [x] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [x] Code cleaned up and commented out code removed.
- [x] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [x] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

**Review by ylli0218 — APPROVED** [review]

**Review by yulongl-ibm — APPROVED** [review]

**Review by mihaela-bornea — APPROVED** [review]
LGTM

File: primeqa/ir/dense/colbert_top/colbert/modeling/hf_colbert_xlmr.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import torch.nn as nn
 
-from transformers import BertPreTrainedModel, BertModel, AutoTokenizer
-from transformers import AutoModel
+from transformers import AutoTokenizer, XLMRobertaModel
```
Comment (avisil) [inline primeqa/ir/dense/colbert_top/colbert/modeling/hf_colbert_xlmr.py:3]: I like this - should've been there before :)

**Review by avisil — DISMISSED** [review]
LGTM

**Comment by avisil** [comment]
I haven't merged - since the branch has conflicts @franzmpub note

**Review by bhavani105 — APPROVED** [review]
LGTM

**Pull request merged**